### PR TITLE
Couple of changes to exercise 2

### PR DESCRIPTION
**Pull request by benprew**
Hey,

Thanks for putting these exercise together, they're fun!

I wanted to expand on the exercise a little bit by adding some more detail to the problem and also adding an artificial slowdown to the mock database call to better simulate a call to a database.  It also highlights the effects of different solutions (ie locking the entire function vs locking just the map/list access)

Finally I was noticed in the comments that it mentioned that it was an LRU cache, but it's implemented as a FIFO cache, so I included some changes to implement an LRU cache.

Let me know if there's any changes you'd like me to make, I'm happy to make any fixes.

Thanks!

### CODE REVIEW

**Comment by loong** [comment]
Hey Ben,

Thank you so much for the PR! Looks great so far. Will give it a closer view over the weekend! 🙏

Greetings from Singapore,
Long

File: 2-race-in-cache/main.go
Comment (loong) [inline 2-race-in-cache/main.go:22]: `Page` is an exported value but has no comment! 

From linter:
```main.go:22:6: exported type Page should have comment or be unexported```

File: 2-race-in-cache/main.go
Comment (loong) [inline 2-race-in-cache/main.go:60]: The if block ends with a return, so else block is not needed.

From linter:
```main.go:47:9: if block ends with a return statement, so drop this else and outdent its block (move short variable declaration to its own line if necessary)```

**Review by loong — CHANGES_REQUESTED** [review]
Hey @benprew!

Thank you so much for the PR! Especially catching my mistake with the LRU. I hope no one used my implementation as a template for their own LRU cache implementation 😅.

I ran `golint` and got some errors. Better to solve them or my friend @ironzeb will give us a bad [GoReportCard](https://goreportcard.com/report/github.com/loong/go-concurrency-exercises)!
```
~/g/s/g/l/g/2-race-in-cache ❯❯❯ golint .
main.go:22:6: exported type Page should have comment or be unexported
main.go:47:9: if block ends with a return statement, so drop this else and outdent its block (move short variable declaration to its own line if necessary)
```

So, I only have a these small nitpicks. Once those are resolved, we can merge!

Greetings from tropical Singapore!

**Comment by benprew** [comment]
Great, thanks for the feedback, I'll make those changes and update the PR.

**Comment by benprew** [comment]
Hey @loong thanks for the feedback, I think my changes address your feedback and the golint warnings.  Let me know if I missed anything.

How's the weather in Singapore?  I'm in Oregon, USA, where it's currently 70F (21C) and sunny, pretty perfect out right now.

File: 2-race-in-cache/main.go
Comment (loong) [inline 2-race-in-cache/main.go:22]: Haha, well that's also a way to pass the linter 😂

**Comment by loong** [comment]
Well, it's an eternal summer here with at least 30°C (86°F) during the day. I grew up in Germany, so this is hot for me (especially with the humidity!). 😅

Went for a hike today!

photo_2020-08-30 20 57 00

**Review by loong — APPROVED** [review]
LGTM! Feel free to merge 🎉

Thank you so much for your contribution! Also, feel free to add your own exercise!

**Comment by benprew** [comment]
Thanks @loong, I'll send another PR if I think of some exercises 👍 .  Also, I don't think I can merge this PR unless you grant me write access to your repo.

PS. that's a great picture, it's so beautiful and serene, more PRs could use pictures like that :)

**Pull request merged**
